internal/nginx/config/generator.go

```diff
@@ -54,11 +56,31 @@ func generate(host state.Host, serviceStore state.ServiceStore) (server, Warning
 			warnings.AddWarning(r.Source, err.Error())
 		}
 
-		loc := location{
-			Path:      g.Path,
-			ProxyPass: generateProxyPass(address),
+		if matchLocationNeeded(r) {
+			path := createPathForMatch(g.Path, r.RuleIdx, r.MatchIdx)
```

Comment:
Good point. I'll change this to use the index to avoid a collision in the future.

Reply:
Changed. Right now the route index is hardcoded since it is always 0. I added a FIXME to change it to the variable once we start handling multiple routes per path.